relay-event-normalization/src/event.rs

```diff
@@ -946,6 +946,7 @@ pub fn normalize_performance_score(
                     );
                 }
             }
+            break; // Stop after the first matching profile.
```

Comment:
This loop has become quite large and contains a few `continue` and `break` statements now, which seems error-prone. I think we could refactor this into a two-step approach that first finds a profile that matches all conditions, then runs calculations with that profile.

```rust



let Some(measurements) = event.measurements.value_mut() { else return };

let Some(profile) = performance_score.profiles.find(|profile| {
    profile.condition.is_some_and(|condition| condition.matches(event))
    && components_match_measurement(&profile.score_components, &measurements)
) { else return };

// ...
```

Reply:
👍 sounds good, i'll follow this up with a refactor pr!